src/Auth/Process/OpaqueSmartID.php

```diff
@@ -429,33 +441,26 @@ private function parseUserId($attribute): string
     /**
      * @param   array  $idpMetadata
      *
-     * @return string
+     * @return  array  IdPs list of emails
      */
-    private function getIdPEmailAddress(array $idpMetadata): string
+    private function getIdPEmailAddress(array $idpMetadata): array
     {
-        $idpEmailAddress = null;
+      $this->logger::debug("[OpaqueSmartID] getIdPEmailAddress: idpMetadata " . var_export($idpMetadata, true));
+        $idpEmailAddress = [];
         if (!empty($idpMetadata['contacts']) && is_array($idpMetadata['contacts'])) {
             foreach ($idpMetadata['contacts'] as $contact) {
                 if (!empty($contact['contactType']) && !empty($contact['emailAddress'])) {
                     if ($contact['contactType'] === 'technical') {
-                        $idpEmailAddress = $contact['emailAddress'];
+                        $idpEmailAddress[] = explode(":", $contact['emailAddress'])[1];
                         continue;
                     } elseif ($contact['contactType'] === 'support') {
-                        $idpEmailAddress = $contact['emailAddress'];
+                        $idpEmailAddress[] = explode(":", $contact['emailAddress'])[1];
                         break;
                     }
                 }
             }
         }
 
-        if (!empty($idpEmailAddress)) {
-            foreach ($idpEmailAddress as &$idpEmailAddressEntry) {
-                if (substr($idpEmailAddressEntry, 0, 7) === "mailto:") {
-                    $idpEmailAddressEntry = substr($idpEmailAddressEntry, 7);
-                }
-            }
-            $idpEmailAddress = implode(";", $idpEmailAddress);
-        }
```

Comment:
If this is removed how do we handle the "mailto:" string in the email addresses now?

Reply:
email addresses do not include colons. If we explode using the colon found between the mailto and the actual email address then the second element of the array will always be the email we want

Reply:
Metadata that follow the spec **MUST** contain `mailto:` as `<EmailAddress>` elements are defined as URIs:

    <EmailAddress> [Zero or More]
    Zero or more elements containing mailto: URIs representing e-mail addresses belonging to the
    contact person

See https://docs.oasis-open.org/security/saml/v2.0/saml-metadata-2.0-os.pdf

However, there are SAML deployments that omit `mailto:` so we need to cater for both scenarios.

If there is an issue with the way that email addresses are handled please create a separate PR.